react-client-app/src/components/MappingTbl.jsx

```diff
@@ -107,20 +107,88 @@ const MappingTbl = (props) => {
+		          	     //get the actual scan_report concept from the ID
+				     //this is hacky and could be more efficient
+				     //get the actualy concept
+				     const scan_report_concept = await useGet(`/scanreportconcepts/${x}`);
+				     //get the object_id
+				     const object_id = scan_report_concept.object_id;
+				     //get the content_type
+				     const content_type = scan_report_concept.content_type;
+				     
+				     let scan_report_value = null;
+				     let scan_report_field = null;
+				     if(content_type==17){ //is a value
+					   scan_report_value = await useGet(`/scanreportvalues/${object_id}`);
+					   scan_report_field = await useGet(`/scanreportfields/${scan_report_value.scan_report_field}`);
+				     }
+				     else{ // (15) - is a field
+					   scan_report_field = await useGet(`/scanreportfields/${object_id}`);
+					   //scan_report_value can be null for the saveMappingRules() function
+				     }
+				     //now need to get the table for saveMappingRules()
+				     const scan_report_table = await useGet(`/scanreporttables/${scan_report_field.scan_report_table}`);
+				     
+				     //saveMappingRules() needs to have the concept attached to the scan_report_concept object
+				     //i.e. needs the details of the concept, not just the ID
+				     const conceptId = scan_report_concept.concept;
+				     const concept = await useGet(`/omop/concepts/${conceptId}`);
+				     //add the retrieved concept
+				     scan_report_concept.concept = concept;
+
+				     //!!warning!!
+				     // - this isnt handled well in the react code
+				     // need to alert or error report about this!
+				     // this happens if a concept has been created,
+				     // and since that creation, the person_id and date event have been removed (or changed)
+				     if (!scan_report_table.person_id || !scan_report_table.date_event) {
+					   let message;
+					   if (!scan_report_table.person_id && !scan_report_table.date_event) {
+						 message = "Please set the person_id and a date_event on the table ";
+					   } else if (!scan_report_table.person_id) {
+						 message = "Please set the person_id on the table ";
+					   } else {
+						 message = "Please set the date_event on the table ";
+					   }
+					   console.log(message);
+					   setLoadingMessage(message);
+					   /*setAlert({
+					     hidden: false,
+					     status: "error",
+					     title: message + scan_report_table.name + " before you add a concept.",
+					     description: "Unsuccessful"
+					     });*/
+				     }
+				     else{
+					   //otherwise we should be good to create the rules
+					   await saveMappingRules(scan_report_concept,scan_report_value,scan_report_table);
+				     }
+				     //console.log(`done with loop ${i}`)
+			       });//end of promises
+			       await Promise.all(promises);
+			       //refresh when done
+			       window.location.reload(true);
+			 })
+		  .catch( (err) =>  console.log(err));
+      }
```

Comment:
an attempt to switch the looping over scan_report_concepts to the react code..

It is difficult as you need to retrieve various data for the scan_report_concept to find the associated scan_report_value and table that is needed by this `saveMappingRules` function